indico/modules/events/registration/fields/sessions.py

```diff
@@ -5,24 +5,33 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import ValidationError, fields, validate
+import json
+from marshmallow import ValidationError, fields, validate, validates_schema
 
 from indico.core.db.sqlalchemy import db
+from indico.core.marshmallow import mm
 from indico.modules.events.registration.fields.base import RegistrationFormFieldBase
 from indico.modules.events.registration.models.registrations import RegistrationData
 from indico.modules.events.sessions.models.blocks import SessionBlock
 from indico.util.i18n import _
 
 
+class SessionsFieldDataSchema(mm.Schema):
+    minimum = fields.Integer(load_default=0, validate=validate.Range(0))
+    maximum = fields.Integer(load_default=0, validate=validate.Range(0))
+    display = fields.String(load_default='expanded', validate=validate.OneOf(['expanded', 'collapsed']))
+
+    @validates_schema(skip_on_field_errors=True)
+    def validate_min_max(self, data, **kwargs):
+        if data['minimum'] and data['maximum'] and data['minimum'] > data['maximum']:
+            raise ValidationError('Maximum value must be less than minimum value.', 'maximum')
+
+
 class SessionsField(RegistrationFormFieldBase):
     name = 'sessions'
     mm_field_class = fields.List
     mm_field_args = (fields.Integer,)
-    setup_schema_fields = {
-        'display': fields.String(load_default='expanded'),
-        'minimum': fields.Integer(load_default=0, validate=validate.Range(0)),
-        'maximum': fields.Integer(load_default=0, validate=validate.Range(0)),
-    }
+    setup_schema_fields = SessionsFieldDataSchema
```

Comment:
I don't think that works, the `setup_schema_fields` is not supposed to be a schema but a dict (when used). This should work as expected though:

```suggestion
    setup_schema_base_cls = SessionsFieldDataSchema
```